sources/web/datalab/polymer/components/datalab-data/datalab-data.ts

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+/// <reference path="../input-dialog/input-dialog.ts" />
+/// <reference path="../item-list/item-list.ts" />
+
+interface Collection {
+  name : string;
+  status : string;
+  tables? : Array<Table>;
+}
+
+interface Table {
+  name : string;
+  status : string;
+}
+
+/**
+ * Data element for Datalab.
+ * Contains a form for initial user input, and a list of tables.
+ */
+class DataElement extends Polymer.Element {
+
+  /** The value the user can enter to filter the collections. */
+  public collectionsFilterValue : string;
```

Comment:
Also, being a public property implies it's an attribute that can be used to customize the element from the host, which doesn't seem like the case here. Should these two be private properties?